### ISSUE TITLE
Tag the WooCommerce Analytics tracking script so that it can be disabled if visitors refuse cookies

### ISSUE THREAD

**Issue by dericleeyy** [Automattic/jetpack#13039]
#### Is your feature request related to a problem? Please describe.
Jetpack WooCommerce Analytics script is not tagged to disable tracking if users refuse cookies, ie through Cookiebot's Cookie consent banner.

#### Describe the solution you'd like

Example, The line that says:

```
<script async src='https://stats.wp.com/s-<id>.js <https://stats.wp.com/s-<id>.js>'>
```

Needs to be modified so it will look like this:

```
<script async src='https://stats.wp.com/s-<id>.js <https://stats.wp.com/s-<id>.js>' type="text/plain" data-cookieconsent="marketing">
```

#### Describe alternatives you've considered

Provide an option to refuse cookies and still use the site.

#### Additional context

Requested from 2165741-zen

**Comment by jeherve** [comment]
I looked into this, and I believe this is something that needs to be implemented within the Cookiebot plugin directly, here:
https://github.com/CybotAS/CookiebotWP/blob/d08bb2fdbe79433e63d5a41883aba0fe782ae0dc/addons/controller/addons/jetpack/jetpack.php

The Cookiebot plugin includes the `Script_Loader_Tag_Interface` that can be used to add extra data attributes to any enqueued files, to mark them as marketing resources for example:
https://github.com/CybotAS/CookiebotWP/blob/8af3203a0496a3302d8c4e922bf166760524cd80/addons/lib/script-loader-tag/script-loader-tag.php#L23

However, for Cookiebot to be able to do that, we need to make a few changes to the way we add the file to sites in the first place. This will happen in #13173.

Once that's merged, we can file an issue with the Cookiebot team to let them know about it.

**Comment by getsnoopy** [comment]
Why does that script exist? And why is it contacting wp.com? Can it be disabled?

**Comment by jeherve** [comment]
@getsnoopy You can find out more about the feature and how it works in the Pull Request where it was introduced: #8296 

It can be disabled if you are not interested in the feature. To do so, you can add the following snippet to [a functionality plugin](https://jetpack.com/support/adding-code-snippets/) on your site:

```php
/**
 * Never load Jetpack's WooCommerce Analytics tool.
 *
 * @param array $tools Array of additional tools loaded by Jetpack.
 */
function jetpackcom_no_woo_analytics( $tools ) {
    $index = array_search( 'woocommerce-analytics/wp-woocommerce-analytics.php', $tools );
    if ( false !== $index ) {
        unset( $tools[ $index ] );
    }
    return $tools;
}
add_filter( 'jetpack_tools_to_include', 'jetpackcom_no_woo_analytics' );
```

**Comment by getsnoopy** [comment]
Thanks. This feature should really be able to be disabled via the Jetpack modules page. It seems like there are many people on the web who have been trying to get rid of it to no avail

**Comment by jeherve** [comment]
@getsnoopy That's a good point, and definitely something worth considering. I'll add this to our to-do list for our next iteration on the feature.

**Comment by jeherve** [comment]
@getsnoopy In #15187, I have now made the feature a module that can disabled from this screen in your dashboard:
`wp-admin/admin.php?page=jetpack_modules`

You can consequently get rid of that code snippet above and disable the module when Jetpack 8.4 is released.

**Comment by getsnoopy** [comment]
@jeherve Quite the quick turnaround. Cheers!

**Comment by jeherve** [comment]
Closing this now, since we've implemented  #13173 to allow third-party plugins to customize that script.
